pkg/cache/cache_test.go

```diff
@@ -1770,6 +1771,303 @@ func TestMatchingClusterQueues(t *testing.T) {
 	}
 }
 
+// TestWaitForPodsReadyCancelled ensures that the WaitForPodsReady call does not block when the context is closed.
+func TestWaitForPodsReadyCancelled(t *testing.T) {
+	scheme := runtime.NewScheme()
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %v", err)
+	}
+	cl := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+	cache := New(cl, WithPodsReadyTracking(true))
+	ctx, cancel := context.WithCancel(context.Background())
+
+	cq := kueue.ClusterQueue{
+		ObjectMeta: metav1.ObjectMeta{Name: "one"},
+	}
+	if err := cache.AddClusterQueue(ctx, &cq); err != nil {
+		t.Fatalf("Failed adding clusterQueue: %v", err)
+	}
+
+	wl := utiltesting.MakeWorkload("a", "").Admit(&kueue.Admission{
+		ClusterQueue: "one",
+	}).Obj()
+	if err := cache.AssumeWorkload(wl); err != nil {
+		t.Fatalf("Failed assuming the workload to block the further admission: %v", err)
+	}
+
+	if cache.podsReadyForAllAdmittedWorkloads(ctx) {
+		t.Fatalf("Unexpected that all admitted workloads are in PodsReady condition")
+	}
+
+	// cancel the context so that the WaitForPodsReady is returns
+	go func() {
+		cancel()
+		cache.CleanUpOnContext(ctx)
+	}()
```

Comment:
```suggestion
	go cancel()
```

`CleanUpOnContext` could be called before.

Reply:
done